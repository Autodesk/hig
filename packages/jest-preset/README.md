# @hig/jest-preset

Jest preset for HIG packages.

## Usage

_Example `package.json`:_

```json
{
  "jest": {
    "preset": "@hig/jest-preset"
  }
}
```
