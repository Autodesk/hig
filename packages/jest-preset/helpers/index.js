/* eslint-disable global-require */
module.exports = {
  takeSnapshotsOf: require("./takeSnapshotsOf")
};
