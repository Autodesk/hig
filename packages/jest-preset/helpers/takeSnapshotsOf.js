const React = require("react");
const renderer = require("react-test-renderer");

module.exports = function takeSnapshotsOf(Component, cases, rendererOptions) {
  cases.forEach(({ desc, props }) => {
    it(desc, () => {
      const vDom = React.createElement(Component, props);
      const tree = renderer.create(vDom, rendererOptions).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
};
