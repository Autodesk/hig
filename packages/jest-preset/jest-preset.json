{
  "collectCoverageFrom": ["**/src/**/*.{js,jsx}"],
  "coverageDirectory": "<rootDir>/coverage",
  "coveragePathIgnorePatterns": [
    "__gemini__",
    "__stories__",
    "docs",
    "examples/redux",
    "node_modules",
    "packages/jest-preset",
    "packages/vanilla",
    "src/playground"
  ],
  "coverageThreshold": {
    "global": {
      "branches": 50,
      "functions": 50,
      "lines": 50,
      "statements": 50
    }
  },
  "moduleNameMapper": {
    "\\.(css|scss|svg)$": "@hig/jest-preset/fileMock.js"
  },
  "modulePaths": ["node_modules"],
  "moduleFileExtensions": ["js", "jsx", "json"],
  "setupFiles": ["raf/polyfill"],
  "setupTestFrameworkScriptFile": "@hig/jest-preset/setupTests.js",
  "testPathIgnorePatterns": [
    "docs",
    "packages/babel-preset",
    "packages/jest-preset",
    "packages/vanilla"
  ],
  "unmockedModulePathPatterns": ["node_modules"]
}
