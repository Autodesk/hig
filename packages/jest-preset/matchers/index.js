require("./toBeConstants");
require("./toImplementHIGInterfaceOf");
require("./toHavePropTypesAndDocgenInfo");
