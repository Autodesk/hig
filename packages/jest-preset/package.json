{
  "private": true,
  "name": "@hig/jest-preset",
  "version": "0.1.0",
  "devDependencies": {
    "babel-core": "^6.26.3",
    "babel-jest": "^23.4.0",
    "enzyme": "^2.9.1",
    "jest": "^21.2.1",
    "jest-enzyme": "^3.8.3",
    "jest-extended": "^0.6.0",
    "raf": "^3.4.0",
    "react-test-renderer": "^15.6.1"
  },
  "eslintConfig": {
    "extends": "@hig"
  },
  "jest": {
    "preset": "@hig/jest-preset"
  }
}
