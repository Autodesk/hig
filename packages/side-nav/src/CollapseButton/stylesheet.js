export default function stylesheet(props, themeData) {
  const { stylesheet: customStylesheet } = props;
  const styles = {
    transform: props.minimized ? "rotate(-90deg)" : "rotate(0deg)",
    height: "100%",
    padding: "6px 10px",
    cursor: "pointer",
  };

  if (customStylesheet) {
    return customStylesheet(styles, props, themeData);
  }

  return styles;
}
