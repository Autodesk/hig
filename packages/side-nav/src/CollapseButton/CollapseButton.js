import React, { Component } from "react";
import PropTypes from "prop-types";
import { css } from "emotion";
import { CaretDownMUI, CaretDownSUI } from "@hig/icons";
import ThemeContext from "@hig/theme-context";
import { memoizeCreateButtonEventHandlers } from "@hig/utils";
import stylesheet from "./stylesheet";

/**
 * @todo Replace with an `IconButton` once it's themeable
 * @todo Deprecate `minimized` prop, and replace it with "pressed" for ARIA consistency.
 * Also, this button shouldn't be concerned with the minimized state of its parent components.
 */
export default class CollapseButton extends Component {
  static propTypes = {
    /** Presents the icon in a minimized state: caret pointing right */
    minimized: PropTypes.bool,
    /** Called when element is clicked */
    onClick: PropTypes.func
  };

  static defaultProps = {
    minimized: false
  };

  createEventHandlers = memoizeCreateButtonEventHandlers();

  render() {
    const { minimized, onClick } = this.props;
    const { handleClick, handleKeyDown } = this.createEventHandlers(onClick);

    return (
      <ThemeContext.Consumer>
        {({ resolvedRoles, metadata }) => (
          <div
            aria-pressed={!minimized}
            className={css(
              stylesheet({ minimized, ...this.props }, resolvedRoles)
            )}
            onClick={handleClick}
            onKeyDown={handleKeyDown}
            role="button"
            tabIndex={0}
          >
            {metadata.densityId === "high-density" ? (
              <CaretDownSUI />
            ) : (
              <CaretDownMUI />
            )}
          </div>
        )}
      </ThemeContext.Consumer>
    );
  }
}
