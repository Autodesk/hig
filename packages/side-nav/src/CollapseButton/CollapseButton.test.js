import { takeSnapshotsOf } from "@weave-design/jest-preset/helpers";

import CollapseButton from "./CollapseButton";

describe("side-nav/CollapseButton", () => {
  describe("snapshot tests", () => {
    takeSnapshotsOf(CollapseButton, [
      {
        desc: "renders with no props",
        props: {},
      },
      {
        desc: "renders minimized",
        props: {
          minimized: true,
        },
      },
      {
        desc: "renders with event handlers",
        props: {
          onClick: function handleClick() {},
        },
      },
    ]);
  });
});
