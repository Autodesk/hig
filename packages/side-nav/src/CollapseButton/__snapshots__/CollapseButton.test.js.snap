// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`side-nav/CollapseButton snapshot tests renders minimized 1`] = `
.emotion-0 {
  fill: #808080;
  pointer-events: none;
}

.emotion-0 > * {
  fill: #808080;
  pointer-events: none;
}

.emotion-1 {
  -webkit-transform: rotate(-90deg);
  -ms-transform: rotate(-90deg);
  transform: rotate(-90deg);
  height: 100%;
  padding: 6px 10px;
  cursor: pointer;
}

<div
  aria-pressed={false}
  className="emotion-1"
  onClick={undefined}
  onKeyDown={undefined}
  role="button"
  tabIndex={0}
>
  <svg
    className="emotion-0"
    height="10px"
    viewBox="0 0 10 10"
    width="10px"
    xmlns="http://www.w3.org/2000/svg"
  >
    <title>
      caret-down-m
    </title>
    <path
      d="M8.71 3.22a.75.75 0 0 0-1.06 0L5 5.87 2.35 3.22a.75.75 0 0 0-1.06 0 .74.74 0 0 0 0 1.06l3.36 3.36a.5.5 0 0 0 .7 0l3.36-3.36a.74.74 0 0 0 0-1.06z"
    />
  </svg>
</div>
`;

exports[`side-nav/CollapseButton snapshot tests renders with event handlers 1`] = `
.emotion-1 {
  -webkit-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
  height: 100%;
  padding: 6px 10px;
  cursor: pointer;
}

.emotion-0 {
  fill: #808080;
  pointer-events: none;
}

.emotion-0 > * {
  fill: #808080;
  pointer-events: none;
}

<div
  aria-pressed={true}
  className="emotion-1"
  onClick={[Function]}
  onKeyDown={[Function]}
  role="button"
  tabIndex={0}
>
  <svg
    className="emotion-0"
    height="10px"
    viewBox="0 0 10 10"
    width="10px"
    xmlns="http://www.w3.org/2000/svg"
  >
    <title>
      caret-down-m
    </title>
    <path
      d="M8.71 3.22a.75.75 0 0 0-1.06 0L5 5.87 2.35 3.22a.75.75 0 0 0-1.06 0 .74.74 0 0 0 0 1.06l3.36 3.36a.5.5 0 0 0 .7 0l3.36-3.36a.74.74 0 0 0 0-1.06z"
    />
  </svg>
</div>
`;

exports[`side-nav/CollapseButton snapshot tests renders with no props 1`] = `
.emotion-1 {
  -webkit-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
  height: 100%;
  padding: 6px 10px;
  cursor: pointer;
}

.emotion-0 {
  fill: #808080;
  pointer-events: none;
}

.emotion-0 > * {
  fill: #808080;
  pointer-events: none;
}

<div
  aria-pressed={true}
  className="emotion-1"
  onClick={undefined}
  onKeyDown={undefined}
  role="button"
  tabIndex={0}
>
  <svg
    className="emotion-0"
    height="10px"
    viewBox="0 0 10 10"
    width="10px"
    xmlns="http://www.w3.org/2000/svg"
  >
    <title>
      caret-down-m
    </title>
    <path
      d="M8.71 3.22a.75.75 0 0 0-1.06 0L5 5.87 2.35 3.22a.75.75 0 0 0-1.06 0 .74.74 0 0 0 0 1.06l3.36 3.36a.5.5 0 0 0 .7 0l3.36-3.36a.74.74 0 0 0 0-1.06z"
    />
  </svg>
</div>
`;
